### PR TITLE
JBPM-7504 Not possible to edit existing Deployment Descriptor config…

### PR DESCRIPTION
**Pull request by bxf12315**
…uration, add edit constants

@cristianonicolai  Updated Please review again

### CODE REVIEW

**Comment by cristianonicolai** [comment]
@bxf12315 overall looks good, but would interesting to see some tests added where applicable. Please aslo use the jbpm jira reference.

**Comment by bxf12315** [comment]
@cristianonicolai  I add two unit test for value modal. But I am not sure the way is right.
Please review it.

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/util/modal/doublevalue/AddDoubleValueModal.java [hunk 1]
```diff
@@ -62,15 +66,21 @@ public void setup(final String headerKey,
         getView().setHeader(translationService.format(headerKey));
         getView().setNameLabel(translationService.format(nameKey));
         getView().setValueLabel(translationService.format(valueKey));
-        super.setup();
```
Comment (cristianonicolai) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/util/modal/doublevalue/AddDoubleValueModal.java]: Isnt doing the same thing, why it needs to be changed?

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/util/modal/doublevalue/AddDoubleValueModal.java [hunk 2]
```diff
@@ -62,15 +66,21 @@ public void setup(final String headerKey,
...
+        superSetup();
     }
 
     public void show(final BiConsumer<String, String> onAdd) {
         this.onAdd = onAdd;
         getView().clearForm();
-        super.show();
```
Comment (cristianonicolai) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/util/modal/doublevalue/AddDoubleValueModal.java]: its probably helpful to keep _super.show_ to make it clear its calling a method from the super class.

Comment (bxf12315) [reply kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/util/modal/doublevalue/AddDoubleValueModal.java]: If keep super.show, the code is not work :doNothing().when(addDoubleValueModal).superSetup();

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/test/java/org/kie/workbench/common/screens/library/client/settings/util/modal/single/AddSingleValueModalTest.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
+/*
+ *
+ *  * Copyright 2017 Red Hat, Inc. and/or its affiliates.
```
[the remark this answers is not in the document]
Comment (bxf12315) [reply kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/test/java/org/kie/workbench/common/screens/library/client/settings/util/modal/single/AddSingleValueModalTest.java]: ok I will change it

**Comment by cristianonicolai** [comment]
@bxf12315 could you please rebase your PR and also squash into a single commit, thanks.

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/util/modal/doublevalue/AddDoubleValueModal.java (hunk 2 above)
Comment (cristianonicolai) [reply kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/util/modal/doublevalue/AddDoubleValueModal.java]: ok, keep _show_ and _setup_.

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/test/java/org/kie/workbench/common/screens/library/client/settings/util/modal/doublevalue/AddDoubleValueModalTest.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        doNothing().when(addDoubleValueModal).superSetup();
+    }
+
+    @Test
+    public void setupTest() {
+        addDoubleValueModal.setup("headKey",
+                                  "nameKey",
+                                  "valueKey");
+
+        verify(view).setHeader(any());
```
Comment (cristianonicolai) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/test/java/org/kie/workbench/common/screens/library/client/settings/util/modal/doublevalue/AddDoubleValueModalTest.java]: test actual values?

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/test/java/org/kie/workbench/common/screens/library/client/settings/util/modal/doublevalue/AddDoubleValueModalTest.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        verify(view).focus();
+    }
+
+    @Test
+    public void showTestWithParam() {
+        addDoubleValueModal.show(any(), any(), any());
+
+        verify(view).clearForm();
+        verify(view).focus();
+        verify(view).setName(any());
```
Comment (cristianonicolai) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/test/java/org/kie/workbench/common/screens/library/client/settings/util/modal/doublevalue/AddDoubleValueModalTest.java]: test actual values?

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/util/modal/doublevalue/AddDoubleValueModal.java (hunk 1 above)
Comment (cristianonicolai) [reply kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/util/modal/doublevalue/AddDoubleValueModal.java]: please update to _setup()_ instead.

Comment (bxf12315) [reply kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/util/modal/doublevalue/AddDoubleValueModal.java]: change superSetup to setup ?

Comment (cristianonicolai) [reply kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/util/modal/doublevalue/AddDoubleValueModal.java]: yes

**Comment by bxf12315** [comment]
@cristianonicolai updated.

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/util/modal/single/AddSingleValueModal.java [hunk 6]
```diff
@@ -56,16 +58,21 @@ public void setup(final String headerKey,
 
         getView().setHeader(translationService.format(headerKey));
         getView().setLabel(translationService.format(labelKey));
-        super.setup();
+        superSetup();
```
Comment (cristianonicolai) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/util/modal/single/AddSingleValueModal.java]: same here, use _setup_

Comment (bxf12315) [reply kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/util/modal/single/AddSingleValueModal.java]: done.

Comment (bxf12315) [reply kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/util/modal/single/AddSingleValueModal.java]: I update unit test and test it . Please review it.

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/util/modal/single/AddSingleValueModal.java [hunk 7]
```diff
@@ -56,16 +58,21 @@ public void setup(final String headerKey,
 
         getView().setHeader(translationService.format(headerKey));
         getView().setLabel(translationService.format(labelKey));
-        super.setup();
+        show();
```
Comment (cristianonicolai) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/util/modal/single/AddSingleValueModal.java]: should be _setup_ in here?

Comment (bxf12315) [reply kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/util/modal/single/AddSingleValueModal.java]: Sorry

**Comment by bxf12315** [comment]
@cristianonicolai For the PR build failure , It is regression 
 bug ,https://github.com/kiegroup/drools/commit/0743d249d15520b4891b1c69fbe5909088e3e461#diff-0fff06c3b351165c25fa1b18767c4ef6L96.
I will raise a pr fix it.

**Comment by bxf12315** [comment]
@cristianonicolai   old solution, so I close it.

**Pull request closed without merging**